### ISSUE TITLE
[primTorch] Reduction references don't return views consistent with their original operators

### ISSUE THREAD

**Issue by mruberry** [pytorch/pytorch#77216]
This occurs because the reduction references model the keepdim arg as a broadcast following the reduction. This makes a lot of sense but causes view inconsistency. For example:

```
a = torch.randn(2, 2)
result = torch.amin(a, keepdim=True, dim=0)
print(result._is_view())
: False

result = refs.amin(a, keepdim=True, dim=0)
print(result._is_view())
: True
```

We should think about a way to resolve this -- making the operations consistent is one option, but maybe primTorch can have a different view behavior in this case.

**Comment by ezyang** [comment]
If you use _unsafe_view it wouldn't report as a view. Since the intermediate doesn't escape this is sound. Cc @albanD @bdhirsh

**Comment by ngimel** [comment]
Yeah but _unsafe_view is not exposed in python, and broadcast_in_dim prim that reduction is calling shouldn't generally use _unsafe_view

**Comment by albanD** [comment]
I think that having it being a view is ok as it is mostly transparent to the user (unless you use private function to check).

**Comment by mruberry** [comment]
From offline discussion with @ngimel:

- this behavior seems fine (per @albanD's point)
- but also we might need _unsafe_view eventually, and we should be open-minded about adding it
